File: Creational-Patterns/C++/Builder/src/hardware/harddisk.cpp
#include "harddisk.hpp"


hardware_infos::HardDisk::HardDisk( std::string hd_type, size_t capacity )
{
    this->hd_type = hd_type;
    this->storage_capacity = capacity;
}

std::string hardware_infos::HardDisk::get_type()
{
    return this->hd_type;
}

size_t hardware_infos::HardDisk::get_capacity()
{
    return this->storage_capacity;
}

File: Creational-Patterns/Python/Builder/hardware/harddisk.py
#coding: utf-8


class HardDisk(object):

    def __init__(self, hd_type : str, hd_capacity : int):
        self.__hd_type = hd_type
        self.__hd_capacity = hd_capacity

    def __str__(self):
        return_str = "== HardDisk == \n"
        return_str += "HardDisk Type : {}\n".format(self.__hd_type)
        return_str += "HardDisk Capacity: {}\n".format(self.__hd_capacity)
        return return_str
